sw/device/lib/dif/dif_i2c.c

```diff
@@ -0,0 +1,624 @@
+
+  return kDifI2cResultOk;
+}
+
+static dif_i2c_result_t irq_bit_for_type(dif_i2c_irq_type_t type,
+                                         uint32_t *bit_index) {
+  switch (type) {
+    case kDifI2cIrqTypeFmtWatermarkUnderflow:
+      *bit_index = I2C_INTR_COMMON_FMT_WATERMARK;
+      break;
+    case kDifI2cIrqTypeRxWatermarkOverflow:
+      *bit_index = I2C_INTR_COMMON_RX_WATERMARK;
+      break;
+    case kDifI2cIrqTypeFmtFifoOverflow:
+      *bit_index = I2C_INTR_COMMON_FMT_OVERFLOW;
+      break;
+    case kDifI2cIrqTypeRxFifoOverflow:
+      *bit_index = I2C_INTR_COMMON_RX_OVERFLOW;
+      break;
+    case kDifI2cIrqTypeNak:
+      *bit_index = I2C_INTR_COMMON_NAK;
+      break;
+    case kDifI2cIrqTypeSclInterference:
+      *bit_index = I2C_INTR_COMMON_SCL_INTERFERENCE;
+      break;
+    case kDifI2cIrqTypeSdaInterference:
+      *bit_index = I2C_INTR_COMMON_SDA_INTERFERENCE;
+      break;
+    case kDifI2cIrqTypeClockStretchTimeout:
+      *bit_index = I2C_INTR_COMMON_STRETCH_TIMEOUT;
+      break;
+    case kDifI2cIrqTypeSdaUnstable:
+      *bit_index = I2C_INTR_COMMON_SDA_UNSTABLE;
+      break;
+    default:
+      return kDifI2cResultBadArg;
+  }
+  return kDifI2cResultOk;
+}
+
+dif_i2c_result_t dif_i2c_irq_get(const dif_i2c_t *i2c, dif_i2c_irq_type_t type,
+                                 bool *flag_out) {
+  if (i2c == NULL || flag_out == NULL) {
+    return kDifI2cResultBadArg;
+  }
+
+  uint32_t index;
+  dif_i2c_result_t err = irq_bit_for_type(type, &index);
+  if (err != kDifI2cResultOk) {
+    return err;
+  }
+
+  *flag_out =
+      mmio_region_get_bit32(i2c->base_addr, I2C_INTR_STATE_REG_OFFSET, index);
+
+  return kDifI2cResultOk;
+}
+
+dif_i2c_result_t dif_i2c_irq_clear(const dif_i2c_t *i2c,
+                                   dif_i2c_irq_type_t type) {
+  if (i2c == NULL) {
+    return kDifI2cResultBadArg;
+  }
+
+  uint32_t index;
+  dif_i2c_result_t err = irq_bit_for_type(type, &index);
+  if (err != kDifI2cResultOk) {
+    return err;
+  }
+
+  // IRQ state registers are write-one-clear.
+  mmio_region_write_only_set_bit32(i2c->base_addr, I2C_INTR_STATE_REG_OFFSET, index);
+
+  return kDifI2cResultOk;
+}
+
+dif_i2c_result_t dif_i2c_irq_set_enabled(const dif_i2c_t *i2c,
+                                         dif_i2c_irq_type_t type,
+                                         dif_i2c_enable_t state) {
+  if (i2c == NULL) {
+    return kDifI2cResultBadArg;
+  }
+
+  uint32_t index;
+  dif_i2c_result_t err = irq_bit_for_type(type, &index);
+  if (err != kDifI2cResultOk) {
+    return err;
+  }
+
+  switch (state) {
+    case kDifI2cEnabled:
+      mmio_region_nonatomic_set_bit32(i2c->base_addr,
+                                      I2C_INTR_ENABLE_REG_OFFSET, index);
+      break;
+    case kDifI2cDisabled:
+      mmio_region_nonatomic_clear_bit32(i2c->base_addr,
+                                        I2C_INTR_ENABLE_REG_OFFSET, index);
+      break;
+    default:
+      return kDifI2cResultBadArg;
+  }
```

Comment:
Use `if/else` instead, then you don't need the default case. There are only ever will be two options `enabled/disabled `. The same for the other similar places.